spec/context.md

```diff
@@ -13,7 +13,66 @@ This document specifies how to compress an Abstract Syntax Tree (or AST) in a gi
 
 ## Grammar
 
-TBD
+A **Grammar** specifies the data structure that we may compress. It may also serve as hints to improve compression.
+
+```typescript
+type Grammar = Set<Declaration>;
+
+class Declaration {
+  name: String;
+  content: Content;
+};
+class Content {
+  optional: bool;
```

Comment:
what is `optional` field for?

Reply:
For optional values. I should document this.